source/agent.js

```diff
@@ -173,7 +178,7 @@ class Agent extends EventEmitter {
 						this._processQueue(name);
 					});
 
-					session.once('remoteSettings', () => {
+					session.once('localSettings', () => {
```

Comment:
Firstly we receive a `SETTINGS` frame from the server, then we receive an acknowledgement of our `SETTINGS` frame. At this point the connection is fully estabilished, we have everything we need to know.